app/src/main/java/com/lagradost/cloudstream3/ui/result/ResultFragmentTv.kt

```diff
@@ -781,25 +781,28 @@ class ResultFragmentTv : Fragment() {
 
                 //    resultEpisodeLoading.isVisible = episodes is Resource.Loading
                 if (episodes is Resource.Success) {
-                    val first = episodes.value.firstOrNull()
-                    if (first != null) {
+
+                    val lastWatchedIndex = episodes.value.indexOfLast { ep -> ep.videoWatchState == VideoWatchState.Watched }
```

Comment:
no? this will get you a wrong result as watchstate is seperate from watched. use getWatchProgress, getDisplayPosition or getRealPosition

Reply:
watch state if smth that you manually toggle

Reply:
or did you mean watch state? because you need to specify that if that is the case

Reply:
I want to get the last index of the "marked as watched" episode, then i return rhe next one so this will be the first unwatched

Reply:
ep.videoWatchState become VideoWatchState.Watched after actual watching or use "Mark as watched"
both works